api/namespace.go

```diff
@@ -70,20 +70,27 @@ func (n *Namespaces) Delete(namespace string, q *WriteOptions) (*WriteMeta, erro
 
 // Namespace is used to serialize a namespace.
 type Namespace struct {
-	Name         string
-	Description  string
-	Quota        string
-	Capabilities *NamespaceCapabilities `hcl:"capabilities,block"`
-	Meta         map[string]string
-	CreateIndex  uint64
-	ModifyIndex  uint64
+	Name                  string
+	Description           string
+	Quota                 string
+	Capabilities          *NamespaceCapabilities          `hcl:"capabilities,block"`
+	NodePoolConfiguration *NamespaceNodePoolConfiguration `hcl:"node_pool_config,block"`
+	Meta                  map[string]string
+	CreateIndex           uint64
+	ModifyIndex           uint64
 }
 
 type NamespaceCapabilities struct {
 	EnabledTaskDrivers  []string `hcl:"enabled_task_drivers"`
 	DisabledTaskDrivers []string `hcl:"disabled_task_drivers"`
 }
 
+type NamespaceNodePoolConfiguration struct {
```

Comment:
These structs are part of our public Go API so we should have a docstring on them it shows up in pkg.go.dev docs.